client/allocrunner/consul_hook_test.go

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package allocrunner
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/nomad/ci"
+	"github.com/hashicorp/nomad/client/allocrunner/interfaces"
+	"github.com/hashicorp/nomad/client/consul"
+	cstructs "github.com/hashicorp/nomad/client/structs"
+	"github.com/hashicorp/nomad/client/widmgr"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/nomad/structs"
+	structsc "github.com/hashicorp/nomad/nomad/structs/config"
+	"github.com/shoenig/test/must"
+)
+
+// statically assert consul hook implements the expected interfaces
+var _ interfaces.RunnerPrerunHook = (*consulHook)(nil)
+
+func testHarness(t *testing.T) *consulHook {
```

Comment:
This is going to sound nitpicky, but this function is visible for all tests in `allocrunner`, so maybe it should be named such that it's scoped to this file?

```suggestion
func consulHookTestHarness(t *testing.T) *consulHook {
```